.github/workflows/antithesis-test-3.5.yml

```diff
@@ -0,0 +1,110 @@
+      email:
+        description: 'Additional email notification recipient (separate with ;)'
+        required: true
+        type: string
+        default: ""
+
+# Declare default permissions as read only.
+permissions: read-all
+
+env:
+  REGISTRY: us-central1-docker.pkg.dev
+  REPOSITORY: molten-verve-216720/linuxfoundation-repository
+
+jobs:
+  build-and-push-and-test:
+    runs-on: ubuntu-latest
+    environment: Antithesis
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Login to Antithesis Docker Registry
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3.4.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: _json_key
+          password: ${{ secrets.ANTITHESIS_CONTAINER_REGISTRY_TOKEN }}
+
+      - name: Build and push config image
+        uses: docker/build-push-action@1dc73863535b631f98b2378be8619f83b136f4a0 # v6.17.0
+        with:
+          context: ./tests/antithesis
+          file: ./tests/antithesis/Dockerfile.config
+          push: true
+          tags: |
+            ${{ env.REGISTRY }}/${{ env.REPOSITORY }}/etcd-config:latest,
+            ${{ env.REGISTRY }}/${{ env.REPOSITORY }}/etcd-config:${{ github.sha }}
+
+      - name: Build and push client image
+        uses: docker/build-push-action@1dc73863535b631f98b2378be8619f83b136f4a0 # v6.17.0
+        with:
+          context: .
+          file: ./tests/antithesis/test-template/Dockerfile
+          push: true
+          tags: |
+            ${{ env.REGISTRY }}/${{ env.REPOSITORY }}/etcd-client:latest,
+            ${{ env.REGISTRY }}/${{ env.REPOSITORY }}/etcd-client:${{ github.sha }}
+
+      - name: Build and push etcd image
+        working-directory: ./tests/antithesis
+        run: |
+          make antithesis-build-etcd-image-3.5
+          export TAG="${{ env.REGISTRY }}/${{ env.REPOSITORY }}/etcd-server:latest"
+          docker tag etcd-server:latest $TAG
+          docker push $TAG
+          export TAG="${{ env.REGISTRY }}/${{ env.REPOSITORY }}/etcd-server:${{ github.sha }}"
+          docker tag etcd-server:latest $TAG
+          docker push $TAG
+
+      - name: Run Antithesis Tests
+        uses: antithesishq/antithesis-trigger-action@b7d0c9d1d9316bd4de73a44144c56636ea3a64ba # main commit on Mar 13, 2025
+        with:
+          notebook_name: etcd
+          tenant: linuxfoundation
+          username: ${{ secrets.ANTITHESIS_WEBHOOK_USERNAME }}
+          password: ${{ secrets.ANTITHESIS_WEBHOOK_PASSWORD }}
+          github_token: ${{ secrets.GH_PAT }}
+          config_image: ${{ inputs.config_image }}
+          images: ${{ inputs.images }}
+          description: ${{ inputs.description }}
+          email_recipients: ${{ inputs.email }}
+          test_name: ${{ inputs.test }}
+          additional_parameters: |-
+            custom.duration = ${{ inputs.duration }}
```

Comment:
@marcus-hodgson-antithesis, on previous meetings you mentioned "source" parameters that would allow us to separate testing for different code branches. How can we pass it? 

I don't see anything like this in https://antithesis.com/docs/using_antithesis/ci/.

Reply:
I think we can track this as a separate issue.  What do you think?

Reply:
It should be a small change, we just need confirmation from @marcus-hodgson-antithesis